src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Tooltip/TagHelperTooltipFactoryBase.cs

```diff
@@ -0,0 +1,200 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.Razor.Tooltip;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Tooltip
+{
+    internal abstract class TagHelperTooltipFactoryBase<T>
+    {
+        protected static readonly Lazy<Regex> ExtractCrefRegex = new(
```

Comment:
Instead of making this protected I'd just wrap "what it does", aka extracting summary content bits, into a descriptive method.